### PR TITLE
feat(#477): rich-text-editor-and-apply

### LINKED ISSUE #477: As an admin, I can edit the informational text shown with the room booking form
### What needs to be done
enhancement: add rich text editor to control description text shown on create booking request page

### PR DESCRIPTION
**Pull request by alecadub**
##### _List all issues this PR resolves_
- closes #477 

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [ ] Documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,5 +97,9 @@
     },
     "jestSonar": {
         "reportPath": "reports/js"
+    },
+    "dependencies": {
+        "tiptap": "^1.32.1",
+        "tiptap-extensions": "^1.35.1"
```
Comment (WindOfRussia) [inline package.json:103]: can we move this to dev dependencies?

**Review by WindOfRussia — CHANGES_REQUESTED** [review]
I really like this simply but effective vue package you picked! I think the original plan was to store these as entries on the settings table instead of creating a new table. You can definitely keep the editor / most of the frontend. But the following modifications should be made:
- move the editor to a new field in the SettingsController
- remove the rawhtml seeder / migration / model / factory
- move the RawHtmlControllerTest into the SettingsControllerTest
- move the RichTextEditor component into the settings page

@alexstojda can you confirm?

**Review by danseremet — CHANGES_REQUESTED** [review]
### Review
I've tested this locally (with the changes from main which you'll need to resolve conflicts, only a small change in the api.php file. 
The editor as it is provided by the tiptap library works correctly.

image

### Expected behavior
When I am submitting a booking request, the text on the left is editable which shouldn't be editable by anyone on that page (I guess you left it there for testing purposes).
The expected behavior is to see these forms with the rich text editor from the settings page instead and only by permitted users.
image
Also, I was not able to test the behavior of the top right button to submit a change of the document which, from looking at the requests sent, does a patch request but results in a 404 error code.

### Requested changes
- Nobody should modify the left side info on the submit a booking request page.
- Have a settings page to modify these booking request page descriptions and save them successfully to be displayed on the actual booking request page.

**Comment by danseremet** [comment]
> I really like this simply but effective vue package you picked! I think the original plan was to store these as entries on the settings table instead of creating a new table. You can definitely keep the editor / most of the frontend. But the following modifications should be made:
> 
> * move the editor to a new field in the SettingsController
> * remove the rawhtml seeder / migration / model / factory
> * move the RawHtmlControllerTest into the SettingsControllerTest
> * move the RichTextEditor component into the settings page
> 
> @alexstojda can you confirm?

Basically, instead of having a generic model RawHtml to save every possible entry of these rich text fields including HTML tags from all pages using it, it would be better to have a per-page XyzHtml model, for example in

File: package.json (hunk 1 above)
Comment (alecadub) [reply package.json:103]: I agree we should move it! 

As for the field, I wasn't sure which one to do, but went directly inside the booking page so the user can see directly what is change will look like before sending it and navigating between pages anyway! 

If we want it inside the settings page, I don't mind I can make the switch this weekend

**Comment by alecadub** [comment]
> ### Review
> I've tested this locally (with the changes from main which you'll need to resolve conflicts, only a small change in the api.php file.
> The editor as it is provided by the tiptap library works correctly.
> 
> image
> 
> ### Expected behavior
> When I am submitting a booking request, the text on the left is editable which shouldn't be editable by anyone on that page (I guess you left it there for testing purposes).
> The expected behavior is to see these forms with the rich text editor from the settings page instead and only by permitted users.
> image
> Also, I was not able to test the behavior of the top right button to submit a change of the document which, from looking at the requests sent, does a patch request but results in a 404 error code.
> 
> ### Requested changes
> * Nobody should modify the left side info on the submit a booking request page.
> * Have a settings page to modify these booking request page descriptions and save them successfully to be displayed on the actual booking request page.

1. There should be a permission check for editing that info, only users with specific permission can do it, but maybe I put the wrong one ^^
2. @WindOfRussia made the same comment about the switch! I'll make it if we want but seriously I find that leaving it directly on the page is better and more user friendly

**Comment by alecadub** [comment]
@danseremet @a-drew made the changes have a look

File: routes/api.php [hunk 2]
```diff
@@ -16,8 +18,9 @@
 |
 */
 
+Route::patch('/booking-setting', [SettingsController::class, 'storeBookingGeneralInformation']);
```
Comment (a-drew) [inline routes/api.php:21]: this route should definitely be inside the auth middleguard

**Review by a-drew — CHANGES_REQUESTED** [review]

File: routes/api.php (hunk 2 above)
Comment (alecadub) [reply routes/api.php:21]: ohh true, forgot to change it after the axios fix

Comment (a-drew) [reply routes/api.php:21]: no worries, smol fix

**Review by a-drew — DISMISSED** [review]

**Comment by a-drew** [comment]
great job on the rework, I'll let dan finish the review.

**Review by danseremet — APPROVED** [review]
I have tested it locally and now it works.
There is a warning on first page load of the settings page because of an "incomingText", but it is just a warning and doesn't seem to affect the functionality anyhow.

**Review by a-drew — APPROVED** [review]

**Pull request merged**